### PR TITLE
add auto_complete

### PR DESCRIPTION
**Pull request by emekaokoli19**
### All Submissions:

- [ ] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [ ] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### New Feature Submission:

- [ ] Does your submission pass the tests? 
- [ ] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.

    a. Give a descriptive title to your PR.

    b. Describe your changes.

    c. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

**Comment by emekaokoli19** [comment]
Could this please be looked at and let me know where to make changes
This pull request is for #345

File: reflex/components/forms/autocomplete.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+"""An AutoComplete omponent."""
```
Comment (picklelo) [inline reflex/components/forms/autocomplete.py:1]: typo in component

File: reflex/components/forms/autocomplete.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    # The placeholder for the input field
+    placeholder: Var[str]
+
+    # Input Form Label to describe the activity or process
+    label: Var[str]
+
+    # For accessibility, you can define a custom color for the highlight color
+    # when user is typing also accept props like yellow.300 based on chakra theme provider
+    highlightItemBg: Var[str]
```
Comment (picklelo) [inline reflex/components/forms/autocomplete.py:25]: Let's make all the props snake_case for consistency, like `highlight_item_bg`. We have logic that will convert them to camelCase for all the components.

File: reflex/components/forms/autocomplete.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    # Function to handle creating new Item
```
Comment (picklelo) [inline reflex/components/forms/autocomplete.py:27]: I don't think we can handle `Callable` props. These will probably need to become event triggers - see here: https://pynecone.io/docs/advanced-guide/wrapping-react. The docs aren't very thorough here so feel free to ask for more clarity

**Review by picklelo — COMMENTED** [review]
Great work, thanks for adding this!

We will also need to add a line in `reflex/components/__init__.py` for the shorthand initializer.

Could you post an example app using the component to test if it works?

**Comment by masenf** [comment]
Closing older issue with no recent updates. Reflex will not be considering further core components based on chakra and will maintain existing chakra components until they can be suitably replaced with Radix or other headless alternatives.

See discussion on #239 for potential future options.

**Pull request closed without merging**